precompiles/utils/src/data.rs

```diff
@@ -187,10 +187,15 @@ impl<'a> EvmDataReader<'a> {
         Ok(data)
     }
 
+    /// Return Option<&[u8]> from a given range for EvmDataReader
+    pub fn get_input_from_range(&self, range: Range<usize>) -> Option<&[u8]> {
+        self.input.get(range)
+    }
+
     /// Move the reading cursor with provided length, and return a range from the previous cursor
     /// location to the new one.
     /// Checks cursor overflows.
-    fn move_cursor(&mut self, len: usize) -> EvmResult<Range<usize>> {
+    pub fn move_cursor(&mut self, len: usize) -> EvmResult<Range<usize>> {
```

Comment:
This is the exact code that was used from moonbeam.
they don't have to make it public because they have defined this function in `mod.rs` which makes this function public inside a module so the don't have to explicitly make it public which is not the case for us.

https://github.com/moonbeam-foundation/moonbeam/blob/c784d124ead5112859b2e1f813f96d4fd6274ebd/precompiles/utils/src/solidity/codec/bytes.rs#L103

The best we can do is to make a wrapper around this function but then we will have to make that public so it will be the same thing, if you have a better way then kindly suggest.

Reply:
Yes, you are right, we can use this function but I would still argue that this just happens to be a wrapper function that is already public and doesn't add to the improvement of design.

But we can have discussion over this later, I have changed this as per your suggestion.

Reply:
The difference is that `move_cursor` manipulates internal fields/attributes.
`read_raw_bytes` is a higher level function that performs a more complex operation, and hides implementation details. 

Maybe not the best read, but first thing Google throws out:
https://doc.rust-lang.org/book/ch17-01-what-is-oo.html#encapsulation-that-hides-implementation-details

The less functionality you expose to the outside, the better.
The higher level the exposed functionality is, the better.
Plus, all the stuff mentioned before (changing existing private to public).

Internet is full of resources to read up about this.